src/preprocessing/passes/bool_to_bv.h

```diff
@@ -46,23 +46,72 @@ class BoolToBV : public PreprocessingPass
     ~Statistics();
   };
 
-  /* Takes an assertion and tries to create more bit-vector structure */
-  Node lowerAssertion(const TNode& a);
+  /** Takes an assertion and attempts to create more bit-vector structure
+      by replacing boolean operators with bit-vector operators.
 
-  /* Tries to lower one node to a width-one bit-vector */
-  void lowerNode(const TNode& n);
+      It passes the force argument down to lowerNode, however it never
+      forces the top-level assertion. There's no point forcing the assertion
+      to be a bit-vector when it will just be converted back into a boolean.
+  */
+  Node lowerAssertion(const TNode& node, bool force = false);
+
+  /** Traverses subterms to turn booleans into bit-vectors using lowerNodeHelper
+   *  Passes the force argument to lowerNodeHelper
+   *  Returns the lowered node
+   */
+  Node lowerNode(const TNode& node, bool force = false);
```

Comment:
(optional) Since this does a traversal, I might call it `recursivelyLowerNode`.

Reply:
No, don't call it `recursively` if it's implemented iteratively -- it might be a bit misleading. And we want to get rid of recursion in general wherever possible.

Reply:
Left it as-is.